exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -322,7 +322,7 @@ private ExecConstants() {
 
   // Controls the flat parquet reader batching constraints (number of record and memory limit)
   public static final String PARQUET_FLAT_BATCH_NUM_RECORDS = "store.parquet.flat.batch.num_records";
-  public static final OptionValidator PARQUET_FLAT_BATCH_NUM_RECORDS_VALIDATOR = new RangeLongValidator(PARQUET_FLAT_BATCH_NUM_RECORDS, 1, Integer.MAX_VALUE);
+  public static final OptionValidator PARQUET_FLAT_BATCH_NUM_RECORDS_VALIDATOR = new RangeLongValidator(PARQUET_FLAT_BATCH_NUM_RECORDS, 1, 65535);
```

Comment:
Should the number be the actual power of 2 -- 2^16 ?  That is - 65536  , without the "minus one" ....

Reply:
I was thinking the same thing, so I double checked and the RangeValidator uses an inclusive range. So I think 65535 is correct.

Reply:
This figure is used to set the `maxRecordsPerBatch` in `RecordBatchSizerManager`; hence every batch sent downstream from the Parquet reader would have at most 64K-1 records. What limit do other operators use ?  
  Though a minor space waste, going "minus one" does have some benefit for variable length columns (see DRILL-5446), where the offsets vector need not double.

Reply:
Yes, you are correct. VL columns use an offset vector which always requires an extra entry; limiting the number of rows to 2^^N-1 guarantees the offset vector will not double in size when the last entry is added.

Reply:
Thanks @Ben-Zvi and @ilooner for the review!

Reply:
Yeah I think your right. I was getting confused between max count and max index value. The Max batch record count used elsewhere is indeed 2^16 as you said. Thx :)